toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -110,13 +119,85 @@ func fetchPackages() (err error) {
 		return
 	}
 
+	// Optional delta build cache hydration
+	if *tryDownloadDeltaRPMs {
+		logger.Log.Info("Attempting to download delta RPMs for build nodes")
+		err = downloadDeltaNodes(dependencyGraph, cloner)
+		if err != nil {
+			err = fmt.Errorf("failed to download delta RPMs: %w", err)
+			return
+		}
+		// Update the package graph with the paths to the delta RPMs we downloaded
+		err = pkggraph.WriteDOTGraphFile(dependencyGraph, *outputGraph)
+		if err != nil {
+			err = fmt.Errorf("failed to write cache graph to file: %w", err)
+			return
+		}
+	}
+
 	// If we grabbed any RPMs, we need to convert them into a local repo
-	if hasUnresolvedNodes {
+	if *tryDownloadDeltaRPMs || hasUnresolvedNodes {
+		logger.Log.Info("Configuring downloaded RPMs as a local repository")
 		err = cloner.ConvertDownloadedPackagesIntoRepo()
 		if err != nil {
 			err = fmt.Errorf("failed to convert downloaded RPMs into a repo:\n%w", err)
 			return err
 		}
+
+		if strings.TrimSpace(*outputSummaryFile) != "" {
+			err = repoutils.SaveClonedRepoContents(cloner, *outputSummaryFile)
```

Comment:
>err = repoutils.SaveClonedRepoContents(cloner, *outputSummaryFile) [](http://example.com/codeflow?start=3&length=66)

Just checking: did we move the `SaveClonedRepoContents` call here because we only need to do it, if anything was downloaded, so we can save some time otherwise?

Reply:
It was originally tied to `resolveGraphNodes()` which only ran conditionally, but I believe it should represent the full state of the cache after all packages are added to it. The delta fetcher will also add nodes, so we want to place this after both have run.

Reply:
Sounds like a good improvement. Especially since it makes `resolveGraphNodes` simpler and focued on one thing. Thanks!